routes/v3/covid-19/apiVaccine.js

```diff
@@ -36,10 +36,11 @@ router.get('/v3/covid-19/vaccine/coverage/countries', async (req, res) => {
 		const numericLastDays = getLastDays(lastdays);
 		const extractedData = countries.map((country) => {
 			const { timeline } = country;
-			const obj = { country: country.country, timeline: {} };
-			Object.keys(timeline).slice(numericLastDays * -1).forEach((date) => {
-				obj.timeline[date] = timeline[date].total;
-			});
+			const obj = {
+				country: country.country,
+				timeline: buildVaccineTimeline(timeline, numericLastDays),
+				timelineList: buildVaccineTimelineList(timeline, numericLastDays)
```

Comment:
as my comment says, I think the `timelineList` field would be better used as a query parameter. So in this case, you could just conditionally set `timeline` based on parameter:
```js
{
    country: country.country,
    timeline: queryParam ? buildVaccineTimeline(timeline, numericLastDays) : buildVaccineTimelineList(timeline, numericLastDays)
}
```

Reply:
We can make this change, however this gives rise to a new problem (of documentation).
To describe different responses in SwaggerUI, we will likely have to use `oneOf` - https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/.
Unfortunately swagger 2.0 does not support this. We would need to change the schema to openapi 3.0.0. If we change the schema, then we will have to rewrite quite a bit of the swagger_v3.json file:
- Change `#/definitions` to `#/components/schemas`
- Change each of these keys: 
```
        "responses": {
          "200": {
            "description": "Status Ok",
            "schema": {
              "$ref": "#/components/schemas/vaccineCoverage"
            }
          }
        }
```
to 
```
        "responses": {
          "200": {
            "description": "Status Ok",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/vaccineCoverage"
                }
              }
```

Would like to hear your thoughts on this - do you see an alternative solution which is simpler? Perhaps we should consider opening a new api version for the newly formatted data instead.